WalletWasabi/Stores/IndexStore.cs

```diff
@@ -38,34 +39,34 @@ public IndexStore(string workFolderPath, Network network, SmartHeaderChain hashC
 
 		private string WorkFolderPath { get; set; }
 		private Network Network { get; }
-		private DigestableSafeMutexIoManager MatureIndexFileManager { get; set; }
-		private DigestableSafeMutexIoManager ImmatureIndexFileManager { get; set; }
+		private DigestableIoManager MatureIndexFileManager { get; set; }
+		private DigestableIoManager ImmatureIndexFileManager { get; set; }
+		private AsyncLock MatureIndexAsyncLock { get; set; } = new AsyncLock();
+		private AsyncLock ImmatureIndexAsyncLock { get; set; } = new AsyncLock();
 		public SmartHeaderChain SmartHeaderChain { get; }
 
 		private FilterModel StartingFilter { get; set; }
 		private uint StartingHeight { get; set; }
 		private List<FilterModel> ImmatureFilters { get; set; }
-		private AsyncLock IndexLock { get; set; }
+		private AsyncLock IndexLock { get; set; } = new AsyncLock();
```

Comment:
I can see that the order of locking in `RemoveAllImmmatureFiltersAsync` is:

1. IndexLock
1. MatureIndexAsyncLock
1. ImmatureIndexAsyncLock

whereas in `TryCommitToFileAsync` it is:

1. MatureIndexAsyncLock
1. ImmatureIndexAsyncLock
1. IndexLock 

This may lead to a deadlock. Do you agree? Am I missing something?

The order of acquiring locks should be documented otherwise, it's deadlock prone. All methods should be have a remark which lock they require, so that it can be verified by a reviewer.

Reply:
ACK